app/src/main/java/com/example/plantsapp/presentation/ui/tasks/adapter/TasksViewHolder.kt

```diff
@@ -1,39 +1,64 @@
 package com.example.plantsapp.presentation.ui.tasks.adapter
 
+import android.graphics.Color
+import android.graphics.Paint
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import com.example.plantsapp.databinding.ItemTaskBinding
 import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.model.TaskWithState
 import com.example.plantsapp.presentation.ui.utils.getIconRes
 import com.example.plantsapp.presentation.ui.utils.getTitleRes
 import com.example.plantsapp.presentation.ui.utils.loadPicture
 
 class TasksViewHolder(
-    private val binding: ItemTaskBinding
+    private val binding: ItemTaskBinding,
+    private val onTaskClick: (Task) -> Unit
 ) : RecyclerView.ViewHolder(binding.root) {
 
-    fun bind(task: Task) {
+    fun bind(taskWithState: TaskWithState) {
         with(binding) {
+            val task = taskWithState.task
 
             tvTaskTitle.text = root.context.getString(
                 task.getTitleRes()
             )
 
             ivTaskIcon.loadPicture(task.getIconRes())
+
+            btnCompleteTask.setOnClickListener {
+                onTaskClick(task)
+            }
+
+            if (taskWithState.isCompleted) {
+                showCompletedTask()
+            }
+        }
+    }
+
+    private fun showCompletedTask() {
```

Comment:
In recyclerView ViewHolders are reused, so when you change view , but do not set process state, your state will be saved on other item, if it happens to redraw them